### DISCUSSION TITLE
Project/User specific dags

### DISCUSSION THREAD

**Discussion by tanisha41** [apache/airflow#36555] [General]
### Description

There is already one option but that is for permissions that what kind of user can view what dags, for example- admin can view all/edit and can do anything, op/public has got only few permissions
Here I am talking about different thing
for example
If rahul is working on a particular project he should access particular dags
vs rajat should be able to dags related to his project or task
they should not have access of either viewing or changing 

### Related issues

I am creating an application on kubernetes cluster and deployed aiflow on that so I think this feature will be helpful

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
See  https://airflow.apache.org/docs/apache-airflow/stable/security/access-control.html#dag-level-permissions. It requires to give permissions to individual dags per user. And you can write synchronization scripts to apply it to multiple DAGs via API.

Easier group/project based permissions will become possible with https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-56+Extensible+user+management and "multi-tenancy" efforts - but it still being worked on.
